examples/twittermap/web/public/javascripts/map/countmap.js

```diff
@@ -2,6 +2,208 @@ angular.module('cloudberry.map')
   .controller('countMapCtrl', function($scope, $compile, cloudberry, cloudberryConfig, MapResultCache,
                                        TimeSeriesCache, moduleManager, cloudberryClient, queryUtil) {
 
+    // Array to store the data for chart
+    $scope.chartData=[];
+    // Map to store the chart data for every polygon
+    $scope.ChartDataMap = new HashMap();
+    // The popup window shown now
+    $scope.popUp = null;
+
+    // return difference of two arrays
+    function arrayDiff (newArray, oldArray) {
+      var diffArray = [], difference = [];
+      for (var i = 0; i < newArray.length; i++) {
+        diffArray[newArray[i]] = true;
+      }
+      for (var j = 0; j < oldArray.length; j++) {
```

Comment:
What will happen if there are duplicates in `oldArray`? Is there an implicit assumption that oldArray must be a set?

Reply:
Yes. When using call `arrayDiff(hasCountMonth,zeroCountMonth)`, both newArray and oldArray has no duplicates.

Reply:
I have added some comment to explain.